### DISCUSSION TITLE
Weird search behavior with stats

### DISCUSSION THREAD

**Discussion by coolassassin** [npm/feedback#685] [Website] [answered]
Hello!

I'm developing [CLI package](https://www.npmjs.com/package/reactcci) for React to make it easier for few years.
And still can't understand the logic of search.

For example, to search my package as user I'm typing in search these words "React create component CLI".
All these words i have in tags. I have exact same words in "description" field in my package.json
I use typescript to make it strict. I use jest with 60% coverage.
I have 406 downloads per month and also 49 stars on github.

But on the other hand, just as example, in this list i see this package before mine:
https://www.npmjs.com/package/add-react

This package way more huge, cuz of bad publish settings. It is 100% js, with no tests, not published for years, so without any maintenance.

It is ok, i can't blame anyone for that. But we can see these stats in search:
<img width="101" alt="image" src="https://user-images.githubusercontent.com/17389477/168464086-e0d7e838-169d-4da3-92c7-980050f97c27.png">  
- 9% popularity
- 53% quality
- 33% maintainence

  
At the same time my package has:  
<img width="103" alt="image" src="https://user-images.githubusercontent.com/17389477/168464149-df3b2e73-4bdb-4c41-a068-b0692f57b3c1.png">  

- 1% popularity
- 63% quality
- 33% maintenance

My package way more popular in fact and has higher quality. But my package on the second page of search, but this on the first.

Could you explain me why. What should i fix to be higher?

P.s.:
Just for example. I see my package with same query on [sixth position](https://yarnpkg.com/?q=create%20react%20component%20cli&p=2).
And first place on [npms.io](https://npms.io/search?q=create+react+component+cli).

**Comment by monishcm** [comment, marked as answer]
Thanks for the feedback. We are investigating our search experience and will include your feedback. We will reach out to you in case we need more information.

**Comment by astoilkov** [comment]
I'm mentioning a related discussion I created that explores a way to make npm search better. The main idea is to give fewer points for popularity and more points for consistency of commits, releases, and issue responses. The longer the period of consistency, the better.

Discussion — https://github.com/npm/feedback/discussions/718
Proposal — https://astoilkov.com/better-npm-search-proposal

**Comment by ganeshkbhat** [comment]
I do not understand the searches metrics so much since I have ot done a deep dive in it. I wish not to comment on that. But here is a question I am raising on the basic download stats:

- I have two packages (both in development stage): [cgijs](https://www.npmjs.com/package/cgijs), [phpcgijs](https://www.npmjs.com/package/phpcgijs).
- The package phpcgjs@1.0.0-beta8 has a dependency cgijs@1.1.0. cgijs@1.1.0 has bee published sometime back. However, phpcgijs@v1.0.0-beta8 has been published a few days (a week) back with a dependency of cgijs. 
- The issue: If I download phpcgijs@v1.0.0-beta8, it has to download cgijs@v1.1.0. Considering last weeks download of 30 for phpcgijs@1.0.0.-beta8, I yet have cgijs@v1.1.0 at 12 downloads a week. (19th to 25th October 2022)
- I am expecting cgijs@v1.1.0 downloads to be at downloads@cgijs@v1.1.0 + downloads@phpcgijs@1.0.0-beta8

This seems to be a gross actual download stats issue. All other stats of downloads and stats based on downloads depends on this calculation. I do not undertand why this was not noticed. I wonder why?

![Web capture_26-10-2022_15300_www npmjs com](https://user-images.githubusercontent.com/3424344/197998416-e20da24b-b4a9-498a-a3a0-c3ff5bac9dc9.jpeg)
![Web capture_26-10-2022_153013_www npmjs com](https://user-images.githubusercontent.com/3424344/197998453-7a8a06ad-cf68-4a08-86f5-076c5c067bb5.jpeg)

**Reply by ljharb** [reply]
Caching could easily account for the discrepancy; it's not reasonable to expect a 1:1 correspondence such that every download a package gets, all of its transitive deps get one as well.
